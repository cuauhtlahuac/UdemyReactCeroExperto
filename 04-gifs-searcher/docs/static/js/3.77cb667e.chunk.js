(this["webpackJsonpgifs-searcher"]=this["webpackJsonpgifs-searcher"]||[]).push([[3],{157:function(e,a,t){"use strict";t.r(a);var n=t(3),c=t(0),i=t.n(c);a.default=function(e){var a=e.gif,t=e.title,s=Object(c.useState)(!1),r=Object(n.a)(s,2),o=r[0],l=r[1];return Object(c.useEffect)((function(){}),[a,o]),i.a.createElement("article",{className:"card animate__animated  animate__flipInX  animate__delay-1s "},i.a.createElement("img",{src:a,alt:t,onLoad:function(){return l(!0)},className:"smooth-image image-".concat(o?"visible":"hidden"," ").concat((!t||" "===t)&&"border-radius-bottom")}),t&&" "!==t&&i.a.createElement("p",null,t))}}}]);
//# sourceMappingURL=3.77cb667e.chunk.js.map