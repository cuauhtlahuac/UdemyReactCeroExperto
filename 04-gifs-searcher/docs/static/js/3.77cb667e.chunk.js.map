{"version":3,"sources":["components/GifGridItem.js"],"names":["GifGridItem","gif","title","useState","imageLoaded","setImageLoaded","useEffect","className","src","alt","onLoad"],"mappings":"wJA0BeA,UAxBK,SAAC,GAAoB,IAAlBC,EAAiB,EAAjBA,IAAKC,EAAY,EAAZA,MAAY,EACCC,oBAAS,GADV,mBAC/BC,EAD+B,KAClBC,EADkB,KAIvC,OADAC,qBAAU,cAAU,CAAEL,EAAKG,IAE1B,6BAASG,UAAU,gEAKlB,yBACCC,IAAKP,EACLQ,IAAKP,EACLQ,OAAQ,kBAAML,GAAe,IAC7BE,UAAS,6BAAwBH,EAC9B,UACA,SAFM,cAEQF,GAAmB,MAAVA,IAAkB,0BAG3CA,GAAmB,MAAVA,GAAkB,2BAAIA","file":"static/js/3.77cb667e.chunk.js","sourcesContent":["import React, { useState, useEffect } from 'react';\n\nconst GifGridItem = ({ gif, title }) => {\n\tconst [ imageLoaded, setImageLoaded ] = useState(false);\n\n\tuseEffect(() => {}, [ gif, imageLoaded ]);\n\treturn (\n\t\t<article className=\"card animate__animated \n\t\tanimate__flipInX \n\t\tanimate__delay-1s\n\t\t\n\t\t\">\n\t\t\t<img\n\t\t\t\tsrc={gif}\n\t\t\t\talt={title}\n\t\t\t\tonLoad={() => setImageLoaded(true)}\n\t\t\t\tclassName={`smooth-image image-${imageLoaded\n\t\t\t\t\t? 'visible'\n\t\t\t\t\t: 'hidden'} ${(!title || title === ' ') && 'border-radius-bottom'}`}\n\t\t\t/>\n\n\t\t\t{(title && title !== ' ') && <p>{title}</p>}\n\t\t</article>\n\t);\n};\n\nexport default GifGridItem;\n"],"sourceRoot":""}