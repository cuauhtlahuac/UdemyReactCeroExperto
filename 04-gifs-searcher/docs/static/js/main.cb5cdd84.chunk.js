(this["webpackJsonpgifs-searcher"]=this["webpackJsonpgifs-searcher"]||[]).push([[0],{155:function(e,t,a){},156:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),c=a(47),i=a.n(c),o=a(3),u=a(29),l=a.n(u),s=a(20),m=a(48),f=a(49),g=a.n(f),p=function(){var e=Object(m.a)(l.a.mark((function e(t){var a,n,r,c,i,o;return l.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return a="https://api.giphy.com/v1/gifs/search?q=".concat(t||"burgers","&limit=20&api_key=").concat("SvFSv1qlzRzuV4jcComxp9kTVC0BL6p4"),e.next=3,fetch(a);case 3:return n=e.sent,e.next=6,n.json();case 6:return r=e.sent,c=r.data,i=c.reduce((function(e,t){var a=t.id,n=t.title,r=t.url,c=t.images.fixed_height;return[].concat(Object(s.a)(e),[{id:a,title:n,link:r,gif:c.webp,webp_size:Number(c.webp_size)}])}),[]),o=g()(i,"webp_size"),e.abrupt("return",o);case 11:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}(),b=Object(n.lazy)((function(){return a.e(3).then(a.bind(null,157))})),d=function(){return r.a.createElement("div",{className:"loading"},"Loading...")},h=function(e){var t=e.category,a=function(e){var t=Object(n.useState)({data:[],loading:!0}),a=Object(o.a)(t,2),r=a[0],c=a[1];return Object(n.useEffect)((function(){p(e).then((function(e){c({data:e,loading:!1})}))}),[e]),r}(t),c=a.data,i=a.loading;return r.a.createElement("section",null,r.a.createElement("h3",{className:"title animate__animated animate__slideInRight"},t),i&&r.a.createElement("p",{className:"loading animate__animated animate__fadeIn animate__repeat-3 animate__faster"},"loading..."),r.a.createElement("section",{className:"image-grid"},c.map((function(e){return r.a.createElement(n.Suspense,{fallback:r.a.createElement(d,null),key:e.id},r.a.createElement(b,e))}))))},y=a(19),E=a(50);var v=function(e){var t=e.setCategories,a=e.setCategoryDictionary,c=e.setSameValue,i=e.categoryDictionary,u=Object(n.useState)(""),l=Object(o.a)(u,2),m=l[0],f=l[1];return r.a.createElement("form",{onSubmit:function(e){e.preventDefault();var n=m.trim();n&&n.length>1&&(a((function(e){return Object(E.a)({},e,Object(y.a)({},n,n))})),i.hasOwnProperty(n)?c(!0):t((function(e){return[n].concat(Object(s.a)(e))})),f(""))},className:"form"},r.a.createElement("label",{htmlFor:"category"},"Category"),r.a.createElement("input",{type:"text",id:"category",value:m,placeholder:"write a favorite gif category",onChange:function(e){return f(e.target.value)}}))},j=function(){var e=Object(n.useState)(["Dragon ball"]),t=Object(o.a)(e,2),a=t[0],c=t[1],i=Object(n.useState)({}),u=Object(o.a)(i,2),l=u[0],s=u[1],m=Object(n.useState)(!1),f=Object(o.a)(m,2),g=f[0],p=f[1];return g?r.a.createElement("div",null,r.a.createElement("h1",null,"You wrote it again, please write other category "),r.a.createElement("button",{onClick:function(){return p(!1)}},"return")):r.a.createElement("section",null,r.a.createElement("h2",{className:"gif-title"},"Gif Searcher"),r.a.createElement(v,{setCategories:c,setCategoryDictionary:s,setSameValue:p,categoryDictionary:l}),r.a.createElement("hr",null),a.map((function(e){return r.a.createElement(h,{category:e,key:e})})))};a(155);i.a.render(r.a.createElement(j,null),document.getElementById("root"))},51:function(e,t,a){e.exports=a(156)}},[[51,1,2]]]);
//# sourceMappingURL=main.cb5cdd84.chunk.js.map