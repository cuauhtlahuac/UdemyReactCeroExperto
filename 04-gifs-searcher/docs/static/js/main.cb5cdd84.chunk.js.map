{"version":3,"sources":["utils/getGifs.js","components/GifGrid.js","hooks/useGetGifs.js","components/AddCategory.js","GifExpertApp.js","index.js"],"names":["getGifs","category","a","url","process","fetch","resp","json","data","imgs","reduce","acc","id","title","fixed_height","images","link","gif","webp","webp_size","Number","sortImgs","sortBy","GifGridItem","lazy","Loader","className","GifGrid","useState","loading","state","setState","useEffect","then","useGetGifs","map","image","fallback","key","AddCategory","setCategories","setCategoryDictionary","setSameValue","categoryDictionary","inputValue","setInputValue","onSubmit","e","preventDefault","trimmedInputValue","trim","length","catDic","hasOwnProperty","cat","htmlFor","type","value","placeholder","onChange","target","GifExpertApp","categories","sameValue","onClick","ReactDOM","render","document","getElementById"],"mappings":"iPACaA,EAAO,uCAAG,WAAMC,GAAN,yBAAAC,EAAA,6DAChBC,EADgB,iDACgCF,GACrD,UAFqB,6BAESG,oCAFT,SAGHC,MAAMF,GAHH,cAGhBG,EAHgB,gBAICA,EAAKC,OAJN,uBAIdC,EAJc,EAIdA,KAEFC,EAAOD,EAAKE,QACjB,SAACC,EAAD,GAKW,IAJTC,EAIQ,EAJRA,GACAC,EAGQ,EAHRA,MACAV,EAEQ,EAFRA,IACUW,EACF,EADRC,OAAUD,aAEX,MAAM,GAAN,mBACIH,GADJ,CAEC,CACCC,KACAC,QACAG,KAAMb,EACNc,IAAKH,EAAaI,KAClBC,UAAWC,OAAON,EAAaK,gBAIlC,IAGKE,EAAWC,IAAOb,EAAM,aA3BR,kBA6BfY,GA7Be,4CAAH,sDCIdE,EAAcC,gBAAK,kBAAM,iCAEzBC,EAAS,kBAAM,yBAAKC,UAAU,WAAf,eAmCNC,EAjCC,SAAC,GAAkB,IAAhB1B,EAAe,EAAfA,SAAe,ECLR,SAAAA,GAAa,IAAD,EACT2B,mBAAS,CAAEpB,KAAM,GAAIqB,SAAS,IADrB,mBAC7BC,EAD6B,KACtBC,EADsB,KAerC,OAZAC,qBACC,WACChC,EAAQC,GAAUgC,MAAK,SAAAxB,GACtBsB,EACC,CAAEvB,KAAMC,EACNoB,SAAS,SAId,CAAE5B,IAGI6B,EDT2BI,CAAWjC,GAA/Bc,EADmB,EACzBP,KAAcqB,EADW,EACXA,QAEtB,OACC,iCACC,wBAAIH,UACJ,iDAGEzB,GAGD4B,GACA,uBAAGH,UACH,+EADA,cASD,6BAASA,UAAU,cACjBX,EAAOoB,KAAI,SAAAC,GAAK,OAChB,kBAAC,WAAD,CAAUC,SAAW,kBAAC,EAAD,MAAaC,IAAKF,EAAMxB,IAC5C,kBAACW,EAAgBa,U,gBEaRG,MA5Cf,YAKI,IAJHC,EAIE,EAJFA,cACAC,EAGE,EAHFA,sBACAC,EAEE,EAFFA,aACAC,EACE,EADFA,mBACE,EACoCf,mBAAS,IAD7C,mBACMgB,EADN,KACkBC,EADlB,KAyBF,OACC,0BAAMC,SArBc,SAAAC,GACpBA,EAAEC,iBAEF,IAAMC,EAAoBL,EAAWM,OAEjCD,GAAqBA,EAAkBE,OAAS,IACnDV,GAAsB,SAAAW,GAAM,sBACxBA,EADwB,eAE1BH,EAAoBA,OAGlBN,EAAmBU,eAAeJ,GACrCP,GAAa,GAEbF,GAAc,SAAAc,GAAG,OAAML,GAAN,mBAA4BK,OAE9CT,EAAc,MAKenB,UAAU,QACxC,2BAAO6B,QAAS,YAAhB,YACC,2BACCC,KAAK,OACL5C,GAAI,WACJ6C,MAAOb,EACPc,YAAY,gCACZC,SA9BsB,SAAAZ,GAAC,OAAIF,EAAcE,EAAEa,OAAOH,YC4BvCI,EAlCM,WACpB,IAD0B,EAEYjC,mBADjB,CAAE,gBADG,mBAElBkC,EAFkB,KAENtB,EAFM,OAG4BZ,mBAAS,IAHrC,mBAGlBe,EAHkB,KAGEF,EAHF,OAIUb,oBAAS,GAJnB,mBAIlBmC,EAJkB,KAIPrB,EAJO,KAM1B,OAAIqB,EAEF,6BACC,gFACA,4BAAQC,QAAS,kBAAMtB,GAAa,KAApC,WAMF,iCACC,wBAAIhB,UAAU,aAAd,gBAEA,kBAAC,EAAD,CACCc,cAAeA,EACfC,sBAAuBA,EACvBC,aAAcA,EACdC,mBAAoBA,IAErB,6BAECmB,EAAW3B,KAAI,SAAAlC,GACf,OAAO,kBAAC,EAAD,CAASA,SAAUA,EAAUqC,IAAKrC,S,OC1B7CgE,IAASC,OAAO,kBAAC,EAAD,MAASC,SAASC,eAAe,U","file":"static/js/main.cb5cdd84.chunk.js","sourcesContent":["import sortBy from 'lodash/sortBy';\nexport const getGifs = async category => {\n\tconst url = `https://api.giphy.com/v1/gifs/search?q=${category ||\n\t\t'burgers'}&limit=20&api_key=${process.env.REACT_APP_GHIPY_KEY}`;\n\tconst resp = await fetch(url);\n\tconst { data } = await resp.json();\n\n\tconst imgs = data.reduce(\n\t\t(acc, {\n\t\t\t\tid,\n\t\t\t\ttitle,\n\t\t\t\turl,\n\t\t\t\timages: { fixed_height },\n\t\t\t   }) => {\n\t\t\treturn [\n\t\t\t\t...acc,\n\t\t\t\t{\n\t\t\t\t\tid,\n\t\t\t\t\ttitle,\n\t\t\t\t\tlink: url,\n\t\t\t\t\tgif: fixed_height.webp,\n\t\t\t\t\twebp_size: Number(fixed_height.webp_size),\n\t\t\t\t},\n\t\t\t];\n\t\t},\n\t\t[],\n\t);\n\t\n\tconst sortImgs = sortBy(imgs, 'webp_size');\n\n\treturn sortImgs;\n};\n","import React, { lazy, Suspense } from 'react';\nimport PropTypes from 'prop-types';\n\nimport { useGetGifs } from 'hooks/useGetGifs';\n\nconst GifGridItem = lazy(() => import('./GifGridItem'));\n\nconst Loader = () => <div className=\"loading\">Loading...</div>;\n\nconst GifGrid = ({ category }) => {\n\tconst { data: images, loading } = useGetGifs(category);\n\n\treturn (\n\t\t<section>\n\t\t\t<h3 className=\n\t\t\t\"title animate__animated\n\t\t\t animate__slideInRight\"\n\t\t\t >\n\t\t\t\t{category}\n\t\t\t</h3>\n\n\t\t\t{loading && (\n\t\t\t\t<p className=\n\t\t\t\t\"loading animate__animated\n\t\t\t\tanimate__fadeIn animate__repeat-3\n\t\t\t\tanimate__faster\"\n\t\t\t\t>\n\t\t\t\t\tloading...\n\t\t\t\t</p>\n\t\t\t)}\n\n\t\t\t<section className=\"image-grid\">\n\t\t\t\t{images.map(image => (\n\t\t\t\t\t<Suspense fallback={ <Loader /> } key={image.id}>\n\t\t\t\t\t\t<GifGridItem {...image} />\n\t\t\t\t\t</Suspense>\n\t\t\t\t))}\n\t\t\t</section>\n\t\t</section>\n\t);\n};\n\nexport default GifGrid;\n\nGifGrid.propTypes = {\n\tcategory: PropTypes.string,\n};\n","import { useState, useEffect } from 'react';\n\nimport { getGifs } from 'utils/getGifs';\n\nexport const useGetGifs = category => {\n\tconst [ state, setState ] = useState({ data: [], loading: true });\n\n\tuseEffect(\n\t\t() => {\n\t\t\tgetGifs(category).then(imgs => {\n\t\t\t\tsetState(\n\t\t\t\t\t{ data: imgs,\n\t\t\t\t\t  loading: false \n\t\t\t\t\t});\n\t\t\t});\n\t\t},\n\t\t[ category ],\n\t);\n\n\treturn state;\n};\n","import React, { useState } from 'react';\nimport PropTypes from 'prop-types';\n\nfunction AddCategory({\n\tsetCategories,\n\tsetCategoryDictionary,\n\tsetSameValue,\n\tcategoryDictionary,\n}) {\n\tconst [ inputValue, setInputValue ] = useState('');\n\n\tconst handleInputValue = e => setInputValue(e.target.value);\n\n\tconst handleSubmit = e => {\n\t\te.preventDefault();\n\n\t\tconst trimmedInputValue = inputValue.trim();\n\n\t\tif (trimmedInputValue && trimmedInputValue.length > 1) {\n\t\t\tsetCategoryDictionary(catDic => ({\n\t\t\t\t...catDic,\n\t\t\t\t[trimmedInputValue]: trimmedInputValue,\n\t\t\t}));\n\n\t\t\tif (categoryDictionary.hasOwnProperty(trimmedInputValue)) {\n\t\t\t\tsetSameValue(true);\n\t\t\t} else {\n\t\t\t\tsetCategories(cat => [ trimmedInputValue, ...cat ]);\n\t\t\t}\n\t\t\tsetInputValue('');\n\t\t}\n\t};\n\n\treturn (\n\t\t<form onSubmit={handleSubmit} className=\"form\">\n\t\t<label htmlFor={\"category\"}>Category</label>\n\t\t\t<input\n\t\t\t\ttype=\"text\"\n\t\t\t\tid={\"category\"}\n\t\t\t\tvalue={inputValue}\n\t\t\t\tplaceholder=\"write a favorite gif category\"\n\t\t\t\tonChange={handleInputValue}\n\t\t\t/>\n\t\t</form>\n\t);\n}\n\nexport default AddCategory;\n\nAddCategory.propTypes = {\n\tsetCategories: PropTypes.func.isRequired,\n};\n","import React, { useState } from 'react';\n\nimport GifGrid from 'components/GifGrid';\nimport AddCategory from 'components/AddCategory';\n\nconst GifExpertApp = () => {\n\tconst initialState = [ 'Dragon ball' ];\n\tconst [ categories, setCategories ] = useState(initialState);\n\tconst [ categoryDictionary, setCategoryDictionary ] = useState({});\n\tconst [ sameValue, setSameValue ] = useState(false);\n\n\tif (sameValue) {\n\t\treturn (\n\t\t\t<div>\n\t\t\t\t<h1>You wrote it again, please write other category </h1>\n\t\t\t\t<button onClick={() => setSameValue(false)}>return</button>\n\t\t\t</div>\n\t\t);\n\t}\n\n\treturn (\n\t\t<section>\n\t\t\t<h2 className=\"gif-title\">Gif Searcher</h2>\n\n\t\t\t<AddCategory\n\t\t\t\tsetCategories={setCategories}\n\t\t\t\tsetCategoryDictionary={setCategoryDictionary}\n\t\t\t\tsetSameValue={setSameValue}\n\t\t\t\tcategoryDictionary={categoryDictionary}\n\t\t\t/>\n\t\t\t<hr />\n\n\t\t\t{categories.map(category => {\n\t\t\t\treturn <GifGrid category={category} key={category} />;\n\t\t\t})}\n\t\t</section>\n\t);\n};\n\nexport default GifExpertApp;\n","import React from 'react';\nimport ReactDOM from 'react-dom';\n\nimport App from 'GifExpertApp';\n\nimport './index.css';\n\nReactDOM.render(<App />, document.getElementById('root'));\n"],"sourceRoot":""}