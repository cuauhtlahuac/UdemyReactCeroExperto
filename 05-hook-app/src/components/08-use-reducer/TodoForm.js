import React from 'react';
import PropTypes from 'prop-types';

import { useForm } from 'hooks/useForm';
import { handleSubmit } from 'utils/todoHandles';

const TodoForm = ({ todos, dispatch }) => {
	const [ { description }, handleInputChange, reset ] = useForm({
		description: '',
	});

	return (
		<React.Fragment>
			<h4>Agregar todo</h4>
			<hr />

			<form onSubmit={e => handleSubmit(e, dispatch, reset, description)}>
				<input
					className="form-control"
					value={description}
					type="text"
					name="description"
					placeholder="Aprender..."
					autoComplete="off"
					onChange={handleInputChange}
				/>

				<button className="btn btn-primary mt-3 btn-block" type="submit">
					Agregar
				</button>
			</form>
		</React.Fragment>
	);
};

TodoForm.propTypes = {
	todos: PropTypes.array,
	dispatch: PropTypes.func,
};

export default TodoForm;
