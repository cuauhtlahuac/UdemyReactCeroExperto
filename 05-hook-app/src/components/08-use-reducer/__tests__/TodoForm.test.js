import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { todoMock } from 'fixtures/todoMock.js';

import TodoForm from '../TodoForm';

describe('TodoApp Test', () => {
	const component = mount(<TodoForm dispatch={jest.fn()} />);

	afterAll(()=>{
		component.unmount();
	})

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});

	it('should have submit button ', () => {
		const button = component.find('[type="submit"]');

		expect(button.exists()).toBe(true);
	});

	test('should call onSubmit', () => {
		const onSubmit = jest.fn(component.find('form').prop('onSubmit'));

		onSubmit({ preventDefault() {} });

		expect(onSubmit).toHaveBeenCalled();
	});

	test('should input change', () => {
		const value = 'Aprender React';

		component.find('input').simulate('change', {
			target: {
				value,
				name: 'description',
			},
		});

		const onSubmit = jest.fn(component.find('form').prop('onSubmit'));

		act(() => {
			onSubmit({ preventDefault() {} });
		});

		expect(onSubmit).toHaveBeenCalled();
	});
});
