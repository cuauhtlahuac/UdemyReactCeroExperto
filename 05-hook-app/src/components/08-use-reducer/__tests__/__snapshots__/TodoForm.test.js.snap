// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp Test should render correctly 1`] = `
Array [
  <h4>
    Agregar todo
  </h4>,
  <hr />,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-primary mt-3 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>,
]
`;
